evm/chains/chain.py

```diff
@@ -48,19 +49,21 @@ class Chain(object):
     VM classes, delegating operations to the appropriate VM depending on the
     current block number.
     """
-    db = None
+    logger = logging.getLogger("evm.chain.chain.Chain")
     header = None
-
+    network_id = None
     vms_by_range = None
 
-    def __init__(self, chaindb, header):
+    def __init__(self, chaindb, header=None):
         if not self.vms_by_range:
             raise ValueError(
                 "The Chain class cannot be instantiated with an empty `vms_by_range`"
             )
 
         self.chaindb = chaindb
         self.header = header
+        if self.header is None:
+            self.header = self.create_header_from_parent(chaindb.get_canonical_head())
```

Comment:
I like this change, however, *if* the `chaindb` doesn't have a canonical head saved, this will raise a `BlockNotFound` error yes?  If so, can we catch this, and re-raise a nicer error that makes it a bit clearer what went wrong.

Reply:
It will actually raise a CanonicalHeadNotFound, as that's what `chaindb.get_canonical_head()` raises.  It could also, in theory, raise a BlockNotFound if there's no header with the hash stored as the head, but `chaindb.set_canonical_head()` shouldn't permit that to happen